### PR TITLE
Rename PolicyBinding API form  `v1alpha1` to `v1beta1`

### PR DESCRIPTION
**Pull request by pjuarezd**
# What does it do?

In preparation for Operator V6 graduates PolicyBinding CRD api `sts.min.io` from `v1alpha1` to `v1beta1` maintaining bakwards compatibility.

This is following the recommendations on the guide https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/

Following step is isolate sts service on it's own deployment

### CODE REVIEW

File: pkg/apis/sts.min.io/v1beta1/doc.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2023, MinIO, Inc.
```
Comment (shtripat) [inline pkg/apis/sts.min.io/v1beta1/doc.go:1]:
```suggestion
// Copyright (C) 2024, MinIO, Inc.
```

File: pkg/apis/sts.min.io/v1beta1/register.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// Copyright (C) 2022, MinIO, Inc.
```
Comment (shtripat) [inline pkg/apis/sts.min.io/v1beta1/register.go:1]:
```suggestion
// Copyright (C) 2024, MinIO, Inc.
```

File: pkg/apis/sts.min.io/v1beta1/types.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2022, MinIO, Inc.
```
Comment (shtripat) [inline pkg/apis/sts.min.io/v1beta1/types.go:1]:
```suggestion
// Copyright (C) 2024, MinIO, Inc.
```

**Review by shtripat — DISMISSED** [review]
Looks fine. Copyright year may be we should keep as 2024.
Also any specific verification steps to be followed?

**Comment by pjuarezd** [comment]
This other PR should help with the Copitight header in the autogenerated code files https://github.com/minio/operator/pull/2120

**Comment by pjuarezd** [comment]
> Looks fine. Copyright year may be we should keep as 2024. Also any specific verification steps to be followed?

There is an automated test already

https://github.com/minio/operator/blob/dae7854a7a1c1c54bacbafca3e5552293baabc3c/.github/workflows/kubernetes-tests.yml#L225-L257

But following documentation steps should still work, I did several manual runs

https://min.io/docs/minio/kubernetes/upstream/developers/sts-for-operator.html

**Review by shtripat — APPROVED** [review]
lgtm

**Pull request merged**